test/behavior/bugs/12043.zig

```diff
@@ -0,0 +1,6 @@
+fn foo(_: anytype) void {}
+
+test {
+    const x = &foo;
+    _ = x;
```

Comment:
I forgot to add, please prefer to make tests actually check that correct runtime behavior happened rather than merely checking that something compiles without crashing. Something like this:

```zig
var ok = false;
fn foo(x: anytype) void {
    ok = x;
}
test {
    const x = &foo;
    x(true);
    try expect(ok);
}
```

Bonus points if the behavior test does not go into the bugs/ directory, but is properly categorized with a nice test name and put with the main behavior tests.